workflow-starfusion/src/main/java/ca/on/oicr/pde/workflows/STARfusionWorkflowClient.java

```diff
@@ -0,0 +1,174 @@
+            queue = getOptionalProperty("queue", "");
+
+            // starfusion
+            starfusionMem = Integer.parseInt(getProperty("starfusion_mem"));
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void setupDirectory() {
+        init();
+        this.addDirectory(dataDir);
+        this.addDirectory(tmpDir);
+        if (!dataDir.endsWith("/")) {
+            dataDir += "/";
+        }
+        if (!tmpDir.endsWith("/")) {
+            tmpDir += "/";
+        }
+    }
+
+    @Override
+    public Map<String, SqwFile> setupFiles() {
+        SqwFile file0 = this.createFile("read1");
+        file0.setSourcePath(read1Fastq);
+        file0.setType(FASTQ_GZIP_MIMETYPE);
+        file0.setIsInput(true);
+        SqwFile file1 = this.createFile("read2");
+        file1.setSourcePath(read2Fastq);
+        file1.setType(FASTQ_GZIP_MIMETYPE);
+        file1.setIsInput(true);
+        return this.getFiles();
+    }
+
+    @Override
+    public void buildWorkflow() {
+
+        /**
+         * STAR-Fusion:
+         */
+        // workflow : read inputs read1 fastq and read2 fastq file; run star-fusion; write the output to temp directory; 
+        // run sequenzaR; handle output; provision files (3) -- .tsv, .tsv, .tsv;
+        Job parentJob = null;
+        this.outDir = this.outputFilenamePrefix + "_output";
+
+        Job starJob = runStarFusion();
+
+        // Provision .seg, .varscanSomatic_confints_CP.txt, model-fit.tar.gz files
+        String fusionPredictionTsv = this.tmpDir + "star-fusion.fusion_predictions.tsv";
+        SqwFile fusionTSV = createOutputFile(fusionPredictionTsv, TXT_METATYPE, this.manualOutput);
+        fusionTSV.getAnnotations().put("STAR_fusion_prediction_tsv", "STAR_fusion");
+        starJob.addFile(fusionTSV);
+
+        String fusionAbridgedTsv = this.tmpDir + "star-fusion.fusion_predictions.abridged.tsv";
+        SqwFile abridgedTSV = createOutputFile(fusionAbridgedTsv, TXT_METATYPE, this.manualOutput);
+        abridgedTSV.getAnnotations().put("STAR_fusion_abridged_tsv", "STAR_fusion");
+        starJob.addFile(abridgedTSV);
+
+        String FFP_coding_effect = this.tmpDir + "star-fusion.fusion_predictions.abridged.coding_effect.tsv";
+        SqwFile codingTSV = createOutputFile(FFP_coding_effect, TXT_METATYPE, this.manualOutput);
+        codingTSV.getAnnotations().put("STAR_fusion_coding_effect_tsv ", "STAR_fusion");
+        starJob.addFile(codingTSV);
+    }
+
+    private Job runStarFusion() {
+        Job starJob = getWorkflow().createBashJob("starfusionjob");
+        Command cmd = starJob.getCommand();
+        cmd.addArgument("export LD_LIBRARY_PATH=" + this.perl + "/lib:$LD_LIBRARY_PATH" + ";");
+        cmd.addArgument("export PERL5LIB=" + this.perl + "/lib:$PERL5LIB" + ";");
+        cmd.addArgument("export PATH=" + this.perl + "/bin:$PATH" + ";");
```

Comment:
Instead of using multiple export statements I would use something like this:

StringBuilder pb = new StringBuilder("export PATH=");
pb.append(this.perl).append("/bin:");
  .append(this.star).append(":");
  .append(this.starfusion).append(":");
  .append(this.tabix),append(":$PATH;");

cmd.addArgument(pb.toString());

(I haven't checked this code, just is just to illustrate the idea. Though it may work as is)